packages/build-tools/src/steps/functions/download_build.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  BuildFunction,
+  BuildStepInput,
+  BuildStepInputValueTypeName,
+  BuildStepOutput,
+} from '@expo/steps';
+
+export function createDownloadBuildFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'download_build',
+    name: 'Download build',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'build_id',
+        required: true,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+    ],
+    outputProviders: [
+      BuildStepOutput.createProvider({
+        id: 'artifact_path',
+        required: false,
+      }),
+    ],
+    command: `
+    BUILD_ID='\${ inputs.build_id }'
+    APPLICATION_ARCHIVE_URL="\${__API_SERVER_URL}v2/artifacts/eas/$BUILD_ID"
```

Comment:
we assume that `__API_SERVER_URL` will always have `/` at the end

isn't it risky? can we do something like `path.join`?